source-git-monorepos/README.md

```diff
@@ -50,21 +89,367 @@ packages:
+paths to dist-git repositories).
+
+In the form above, the command would update only the dist-git repositories
+which are specified on the command line. If any of the `DIST_GIT` arguments
+specified is not a `downstream_package_name` in `.distro/source-git.yaml`, the
+command would fail. This would be an early check, before any other change is
+done.
+
+An alternative form of the command could allow updating multiple dist-git
+repositories found in `--dist-git-store`. The command would figure out
+dist-git subdirectory names in `--dist-git-store`, based on
+`downstream_package_name`s found in `.distro/source-git.yaml`. If any of them
+is missing, the command would fail early.
+
+```
+SYNOPSIS
+       packit source-git update-dist-git [OPTIONS] --dist-git-store PATH SOURCE_GIT
+```
+
+Example:
+
+```shell
+$ packit source-git update-dist-git --dist-git-store rpms/ src/copr
+```
+
+### `packit source-git update-source-git`
+
+Let's not change the interface of this command and don't support the usecase
+to update a single source-git monorepo from multiple dist-git repositories in
+order to avoid complications. Expect that multiple commands are going to be
+executed if updates need to be taken from multiple dist-git repositories.
+
+On the other hand, the command needs to be updated to support a source-git
+repo as a target when the source-git repo is a monorepo. In this case only the
+paths of the sub-package need to be considered during the update. `DIST_GIT`
+should match one of the `downstream_package_name`s in `.distro/source-git.yaml`.
+
+### `packit source-git status`
+
+The command should check the sync status of a sub-package from a source-git
+monorepo and the corresponding dist-git repo. `downstream_package_name` should
+match `basename DIST_DIS`.
+
+### Hardly
+
+When opening an MR in a source-git monorepo, Hardly should figure out which
+dist-git packages are going to be impacted by the changed (by looking at
+`package_paths`), and open mirror-MRs in the dist-git repositories of those
+packages.
+
+CI results from all mirror-MRs should be synced back to the source-git MR.
+
+**Open question**: Is the behavior above good enough? Can there be source-git
+changes happening outside the package paths, but which influence the dist-git
+content of the package?
+
+## Impact on Packit
+
+All things that follow assume that [the `metadata` key was dropped].
+
+Additionally to what was presented above, a new property of job-objects in
+`.packit.yaml` needs to be introduced, to be used to specify which packages
+the job should handle when triggered. The property would be called `packages`
+and would be a list of package-object names. If a job-object has no such key,
+all packages of a monorepo should be handled.
+
+Example:
+
+```yaml
+upstream_project_url: https://pagure.io/copr/copr.git
+packages:
+  copr-backend:
+    specfile_path: backend/copr-backend.spec
+    package_paths:
+      - backend
```

Comment:
In Copr we by default search for `<package_path>/<package_name>.spec`, to simplify the config.

Reply:
I've made a proposal for this: see the "Package and specfile discovery" chapter.